### ISSUE TITLE
Allow an empty block for a branch in a when statement with `BracesOnWhenStatements`

### ISSUE THREAD

**Issue by eygraber** [detekt/detekt#6039]
## Expected Behavior

For the following configuration of the `BracesOnWhenStatements` rule, I would expect to be allowed to use an empty block for a branch in a when statement:

```yaml
BracesOnWhenStatements:
  active: true
  singleLine: 'necessary'
  multiLine: 'necessary'
```

## Current Behavior

It is not allowed.

```kotlin
enum class Action {
  DoSomething,
  DoNotDoSomething,
  DoNotDoSomethingWithComment
}

fun test(action: Action) {
  when(action) {
    DoSomething -> println("Doing something") // not a violation
    DoNotDoSomething -> {} // is a violation
    DoNotDoSomethingWithComment -> { /* we're not doing something */ } // is a violation
  }
}
```
